ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -524,7 +526,10 @@ private void addDatasetVersion(BlobProperties blobProperties, RestRequest restRe
         restResponseChannel.setHeader(RestUtils.Headers.TARGET_DATASET_NAME, datasetName);
         restResponseChannel.setHeader(RestUtils.Headers.TARGET_DATASET_VERSION, datasetVersionRecord.getVersion());
         long newExpirationTimeMs = datasetVersionRecord.getExpirationTimeMs();
-        blobProperties.setTimeToLiveInSeconds(newExpirationTimeMs);
+        if (expirationTimeMs != Utils.Infinite_Time || expirationTimeMs != newExpirationTimeMs) {
```

Comment:
Can you add comment to explain the difference between the dataset creation time, expiration and ttl?

Reply:
Updated.